src/main/java/frc/robot/RobotContainer.java

```diff
@@ -132,84 +108,59 @@ public class RobotContainer {
-        cubeTopNode = new SetElevatorPositionCmd(elevator, ElevatorConstants.CUBE_TOP_NODE_HEIGHT);
-        coneMidNode = new SetElevatorPositionCmd(elevator, ElevatorConstants.CONE_MID_NODE_HEIGHT);
-        cubeMidNode = new SetElevatorPositionCmd(elevator, ElevatorConstants.CUBE_MID_NODE_HEIGHT);
-        coneLowNode = new SetElevatorPositionCmd(elevator, ElevatorConstants.CONE_LOW_NODE_HEIGHT);
-        cubeLowNode = new SetElevatorPositionCmd(elevator, ElevatorConstants.CUBE_LOW_NODE_HEIGHT);
-
-        setElevatorSpeedUp = new RunCommand(() -> elevator.setSpeed(0.2), elevator);
-        setElevatorSpeedDown = new RunCommand(() -> elevator.setSpeed(-0.2), elevator);
-        stopElevator = new InstantCommand(() -> elevator.setSpeed(0), elevator);
-        collectPiece = new CollectPieceCmd(intake);
-        dropCone  = new InstantCommand(() -> intake.drop(), intake);
-        retractElevator = new SetElevatorPositionCmd(elevator, Constants.ElevatorConstants.MIN_ELEVATOR_HEIGHT);
-
-        noSpin = new RunCommand(() -> intake.setMotor(0), intake);
-        spinIn = new RunCommand(() -> intake.setMotor(Constants.IntakeConstants.INTAKE_IN_SPEED), intake);
-        spitOut = new RunCommand(() -> intake.setMotor(Constants.IntakeConstants.INTAKE_OUT_SPEED), intake);    
     
-        moveArmUp = new InstantCommand(() -> {arm.setTargetAngle(Math.PI/4);});
-        moveArmDown = new InstantCommand(() -> {arm.setTargetAngle(-Math.PI/4 * 3);});
-        armDefaultCmd = new SetArmAngleCmd(arm);
-        moveArmHalfway = new InstantCommand(() -> {arm.setTargetAngle(-Math.PI/4);});
+        setUpSubsystems();
+        setUpAutonChooser();
+        setUpConeCubeCommands();
+        configureButtonBindings();
+        setDefaultCommands();
+        simulationMechanisms();
 
-        changeMode = new InstantCommand(() -> {coneMode = !coneMode;});
+    }
 
-        // changeToConeMode = new InstantCommand(() -> {coneMode = true;});
-        // changeToCubeMode = new InstantCommand(() -> {coneMode = false;});
+    private void configureButtonBindings() {
+       
+        new JoystickButton(xbox, Button.kA.value).onTrue(changeMode);
+       
+        // new JoystickButton(xbox,XboxMappingToJoystick.A_BUTTON).onTrue(new InstantCommand(() -> {coneMode = true;}));
+        // new JoystickButton(xbox,XboxMappingToJoystick.B_BUTTON).onTrue(new InstantCommand(() -> {coneMode = false;}));
 
-        placePieceTop = new ConditionalCommand(coneTopNode, cubeTopNode, () -> coneMode);
-        placePieceMid = new ConditionalCommand(coneMidNode, cubeMidNode, () -> coneMode);
-        placePieceLow = new ConditionalCommand(coneLowNode, cubeLowNode, () -> coneMode);
 
-        engage = new EngageCmd();
+        // if coneMode true, set elevator to cone mode for top node
+        new JoystickButton(xbox, Button.kX.value).onTrue(new ConditionalCommand(coneTopNode, cubeTopNode, () -> coneMode));
```

Comment:
This crashes in sim because coneTopNode can't be put into another command because it was already put into a command (placePieceTop) (isn't that command what you want here?)

Reply:
You can (of course) use the simulator to check all commands work properly. Also: great job with everything, it looks really neat now